src/JWT/Jwt.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+
+namespace JWT
+{
+    /// <summary>
+    /// A class that represent a JWT
+    /// </summary>
+    internal class Jwt
+    {
+        /// <summary>
+        /// A array that have length of 3, because the JWT have 3 Parts
+        /// </summary>
+        private readonly string[] _parts;
+
+        /// <summary>
+        /// Creates a new Instance of JWT from a JWT-String
+        /// </summary>
+        /// <param name="token">The JWT as string</param>
+        public Jwt(string token): this(SplitToken(token)) { }
+
+        /// <summary>
+        /// Create a new Instance of a JWT from a part Array of an JWT
+        /// </summary>
+        /// <param name="parts">The parts as Array</param>
+        public Jwt(string[] parts) => _parts = parts;
+
+        /// <summary>
+        /// gets the Header of an JWT
+        /// </summary>
+        public string Header => _parts[(int)JwtParts.Header];
+        /// <summary>
+        /// gets the Payload of an JWT
+        /// </summary>
+        public string Payload => _parts[(int)JwtParts.Payload];
+
+        /// <summary>
+        /// gets the Signature of an JWT
+        /// </summary>
+        public string Signature => _parts[(int)JwtParts.Payload];
+
+        /// <summary>
+        /// gets the Parts of an JWT
+        /// </summary>
+        public string[] Parts => _parts.ToArray();
+
+        /// <summary>
+        /// Split a JWT-String in to its parts
+        /// </summary>
+        private static string[] SplitToken(string token)
```

Comment:
SplitToParts since that's what it is, right?

Reply:
yes.. sorry. can we leave it that way?

Reply:
Looks from the perspective of internal reader, preferably not that technical. Because this is almost the only documentation we have here. Which one your granny would understand? I'm not a native English speaker so it's little hard me to tell.

Also we can ask @nbarbettini to take a look and suggest, please.

Reply:
Good idea. I am also not a native english speaker 😃